doc/memory.rst

```diff
@@ -324,26 +324,65 @@ Gotchas
     >>> print(sin(0))
     0.0
 
-* **caching methods**: you cannot decorate a method at class definition,
-  because when the class is instantiated, the first argument (self) is
-  *bound*, and no longer accessible to the `Memory` object. The following
-  code won't work::
+* **caching methods: memory is designed for pure functions and it is
+  not recommended to use it for methods**. If you want to use cache
+  inside a class the recommended pattern is to cache a pure function
+  and use the cached function inside your class, i.e. something like
+  this::
+
+    @mem.cache
+    def compute_func(arg1, arg2, arg3):
+        # long computation
+        return result
+
 
     class Foo(object):
+        def __init__(self, args):
+            self.data = None
 
-        @mem.cache  # WRONG
-        def method(self, args):
-	    pass
+        def compute(self):
```

Comment:
I actually now prefer to use joblib not as an decorator, but as an explicit function transformation, and I would remove the decorator on compute_func and write the following:

self.data = mem.cache(compute_func)(self.arg1, self.arg2, 40)